packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -86,7 +86,13 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
     if (e.pointerType === 'keyboard' && isNonContiguousSelectionModifier(e)) {
       manager.toggleSelection(key);
     } else {
-      manager.select(key, e);
+      let eventType: SelectEventType = {};
+      if (e && e.shiftKey) {
+        eventType.extend = true;
+      } else if (e && (((e.metaKey && !isWindows()) || e.pointerType === 'touch' || e.pointerType === 'virtual') || (e.ctrlKey && isWindows()))) {
+        eventType.toggle = true;
+      }
+      manager.select(key, eventType);
```

Comment:
Maybe we should just call the individual methods directly from here rather than going through the `select` method?

Reply:
I liked the idea of select handling the specifics of how SelectionManager worked so consumers don't have to figure out how to change selection based on the selection mode and other selection parameters.

Reply:
but isn't that kind of what you're doing with
```
export interface SelectEventType {
  extend?: boolean,
  toggle?: boolean
}
```
also, if we have any other use cases or the use cases changes, then it might be more flexible to just call them directly

Reply:
`Select` makes choices about selection based on single, multiple, disallowEmptySelection, and no selection with these key events being edge cases factored into the previous selection choices. All this logic about selection seems like details SelectionManager should hide with aria making suggestions based on keys and click events.